src/backend/joanie/core/models/courses.py

```diff
@@ -233,27 +233,34 @@ def signature_backend_references_to_sign(
                 submitted_for_signature_on__isnull=False,
                 student_signed_on__isnull=False,
                 order__organization=self,
-            ).values_list("signature_backend_reference", flat=True)
+            ).values_list("id", "signature_backend_reference")
         )
         if contracts_ids and len(contracts_to_sign) != len(contracts_ids):
             raise exceptions.NoContractToSignError(
                 "Some contracts are not available for this organization."
             )
-        return contracts_to_sign
+
+        contract_ids = [contract[0] for contract in contracts_to_sign]
+        references = [contract[1] for contract in contracts_to_sign]
+
+        return (references, contract_ids)
```

Comment:
Some test would be slightly edited, but we can simplify with:
```suggestion
        contract_ids, references = zip(*contracts_to_sign)
        return references, contract_ids
```
Or even 
```suggestion
        return tuple(zip(*contracts_to_sign))
```